File: 백준/Gold/1253. 좋다/좋다.cc
#include<bits/stdc++.h>
using namespace std;
int main()
{
	int N;
	cin>>N;
	vector<long long> A(N);
	for(int i=0;i<N;i++)
	{
		cin>>A[i];
	}
	sort(A.begin(),A.end());
	int result=0;
	
	for(int i=0;i<N;i++)
	{
		int left=0;
		int right=N-1;
		while(left<right)
		{
			if(left==i)
			{
				left++;
				continue;
			}
			if(right==i)
			{
				right--;
				continue;
			}
			long long sum = A[left] + A[right];
			
			if(sum==A[i])
			{
				result++;
				break;
			}
			else if(sum<A[i])
			{
				left++;
			}
			else
			{
				right--;
			}
		}
	}
	cout<<result<<"\n";
	return 0;
}

File: 백준/Gold/1253. 좋다/좋다.py
n=int(input().strip())
nums=list(map(int,input().split()))
nums.sort()
ret=0

for i in range(n):
    left=0
    right=n-1

    while left<right:
        if left==i:
            left+=1
            continue
        if right==i:
            right-=1
            continue

        sum_val=nums[left]+nums[right]

        if sum_val==nums[i]:
            ret+=1
            break
        elif sum_val<nums[i]:
            left+=1
        else:
            right-=1

print(ret)